data-prepper-api/src/main/java/com/amazon/dataprepper/model/configuration/PipelineModel.java

```diff
@@ -64,6 +65,9 @@ else if (preppers != null) {
     @JsonInclude(JsonInclude.Include.NON_NULL)
     private final PluginModel buffer;
 
+    @JsonProperty("router")
+    private List<PipelineConditionalRoute> router;
```

Comment:
The [RFC](#1007) calls for:

```
router:
  - myRouteA: '/some/conditional'
  - myRouteA: '/some/conditional'
```

I tried to use Jackson to create a `Router` class as an abstraction of the whole router concept. It could then have `routes` (which are not exposed in the YAML model, only the Java model) But, I hit quite a few limitations with Jackson that prevented it from working the way I would have liked.

There is a bit of discussion in that RFC, but overall the concept of a router seemed most appropriate. We could rename it to `routes`. We could also consider making a model like the following:

```
router:
  routes:
    - myRouteA: '/some/conditional'
    - myRouteA: '/some/conditional'
```

This last approach actually lines up with what I was trying to do with the Java model. I just know know if that extra nesting is really valuable to pipeline authors.